napari/layers/_vectors_layer/model.py

```diff
@@ -443,13 +332,26 @@ def _set_view_slice(self):
         """Sets the view given the indices to slice with."""
 
         vertices = self._mesh_vertices
-        faces = self._mesh_triangles
+
+        if len(self.data) == 0:
+            faces = []
+        elif self.ndim > 2:
+            matches = np.all(self.indices[:-2] == self.data[:, 0, :-2], axis=1)
+            matches = np.where(matches)[0]
```

Comment:
what are "matches?"

Reply:
Vectors in the currently viewed plane. I've added a comment